### PR TITLE
Pre and Post submit jobs for Athens

### PR DESCRIPTION
**Pull request by bnrjee**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by TerryHowe — APPROVED** [review]

File: jobs/aws/eks-distro-build-tooling/athens-postsubmits.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      containers:
+      - name: athens
+        image: 316434458148.dkr.ecr.us-west-2.amazonaws.com/eks-distro/builder:e35d4971088ca37db784caf525373f7342126bf3
+        command:
+        - bash
+        - -c
+        - >
+          make release -C athens
+          &&
+          touch /status/done
```
Comment (abhay-krishna) [inline jobs/aws/eks-distro-build-tooling/athens-postsubmits.yaml:41]: Can you add the Livenessprobe tests and buildkit container similar to this - https://github.com/aws/eks-distro-prow-jobs/blob/main/jobs/aws/eks-distro-build-tooling/eks-distro-base-postsubmits.yaml

We need livenessprobe to ensure prowjobs always end, and the buildkit container to be able to use buildctl.

File: jobs/aws/eks-distro-build-tooling/athens-postsubmits.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+postsubmits:
+  aws/eks-distro-build-tooling:
+  - name: athens-postsubmit
+    always_run: false
+    run_if_changed: "athens/*"
```
Comment (micahhausler) [inline jobs/aws/eks-distro-build-tooling/athens-postsubmits.yaml:19]: Can you do the same for the presubmit?
```suggestion
    run_if_changed: "projects/gomods/athens/*"
```

File: jobs/aws/eks-distro-build-tooling/athens-postsubmits.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    spec:
+      serviceaccountName: postsubmits-build-account
+      containers:
+      - name: build-container
+        image: 316434458148.dkr.ecr.us-west-2.amazonaws.com/eks-distro/builder:e35d4971088ca37db784caf525373f7342126bf3
+        command:
+        - bash
+        - -c
+        - >
+          make release -C athens
```
Comment (micahhausler) [inline jobs/aws/eks-distro-build-tooling/athens-postsubmits.yaml:39]: Same for presubmit
```suggestion
          make release -C projects/gomods/athens
```

**Review by micahhausler — CHANGES_REQUESTED** [review]

**Review by micahhausler — APPROVED** [review]
/approve
/lgtm

**Pull request merged**